src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt

```diff
@@ -86,22 +94,50 @@ class TestCaseDisplayService(private val project: Project) {
             // fix Windows line separators
             val testCodeFormatted = testCode.replace("\r\n", "\n")
 
+            // Add checkbox
             val checkbox = JCheckBox()
             checkbox.isSelected = true
             testCasePanel.add(checkbox, BorderLayout.WEST)
 
+            // Toggle coverage when checkbox is clicked
             checkbox.addItemListener {
                 project.messageBus.syncPublisher(COVERAGE_SELECTION_TOGGLE_TOPIC)
                     .testGenerationResult(testName, checkbox.isSelected, editor)
             }
 
+            // Add editor
             val document = EditorFactory.getInstance().createDocument(testCodeFormatted)
             val textFieldEditor = EditorTextField(document, project, JavaFileType.INSTANCE)
-
             textFieldEditor.setOneLineMode(false)
-
             testCasePanel.add(textFieldEditor, BorderLayout.CENTER)
 
+            // Add top buttons
+            val topButtons = JPanel()
+            topButtons.layout = FlowLayout(FlowLayout.TRAILING)
+
+            // Add "Reset" button
+            val resetButton = JButton("Reset")
+            resetButton.isEnabled = false
+            resetButton.addActionListener {
+                WriteCommandAction.runWriteCommandAction(project) {
+                    document.setText(testCodeFormatted)
+                    resetButton.isEnabled = false
+                    textFieldEditor.border = JBUI.Borders.empty()
+                }
+            }
+            // enable reset button when editor is changed
+            document.addDocumentListener(object : DocumentListener {
+                override fun documentChanged(event: DocumentEvent) {
+                    resetButton.isEnabled = true
+                    val service = TestGenieSettingsService.getInstance().state
+                    val borderColor = Color(service!!.colorRed, service.colorGreen, service.colorBlue)
+                    textFieldEditor.border = BorderFactory.createLineBorder(borderColor)
```

Comment:
So, the actual `Diff`text highlighting was not possible?

Reply:
I tried to make it highlight the changed text, but it appears that the small editors that we use in the tool window don't actually support highlighting as nothing appears when I call any of the highlighting methods.

Reply:
I see